packages/netsuite-adapter/src/change_validators/unique_fields.ts

```diff
@@ -0,0 +1,235 @@
+
+const getChangeNestedField = (
+  change: ChangeDataType,
+  field: string
+) : string => resolvePath(change, change.elemID.createNestedID(field))
+
+const getCustomRecordRestrictedData = async ({ elemID, elementsSource }: GetterParams):
+  Promise<string[]> => {
+  const element = await elementsSource.get(elemID)
+  if (!isObjectType(element) || !isCustomRecordType(element)) {
+    return []
+  }
+
+  return Object.keys(element.fields)
+    .filter(field => field.startsWith(CUSTOM_RECORD_FIELD_PREFIX))
+    .map(field => element.fields[field].annotations.scriptid)
+}
+
+/* If you wish validate another type's unique fields, just:
+      1. Add its type definition (of your choice) to RestrictedTypes.
+      2. Define its RestrictedTypeGetters.
+      3. Update the getters map.
+      4. Update getRestrictedType to match the new type as well. */
+
+type RestrictedTypes = typeof SAVED_SEARCH_TYPE | typeof FINANCIAL_LAYOUT_TYPE | typeof CUSTOM_RECORD_FIELD_TYPE
+
+const savedSearchGetters: RestrictedTypeGetters = {
+  getChangeRestrictedField: change => getChangeNestedField(change, FIELD_DEFAULT_NAME),
+  getSourceRestrictedFields: params => getNestedField(params, FIELD_DEFAULT_NAME),
+  getMessage: () => 'A Saved Search with that title already exists',
+  getDetailedMessage: field => `Can't deploy the Saved Search: ${field} because there is already a Saved Search with that title in the target environment.`
+    + ' Please select a Saved Search with a unique title.',
+}
+
+const financialLayoutGetters: RestrictedTypeGetters = {
+  getChangeRestrictedField: change => getChangeNestedField(change, NAME_FIELD),
+  getSourceRestrictedFields: params => getNestedField(params, NAME_FIELD),
+  getMessage: () => 'A Financial Layout with that name already exists',
+  getDetailedMessage: (field: string) => `Can't deploy the Financial Layout: ${field} because there is already a Financial Layout with that name in the target environment.`
+    + ' Please select a Financial Layout with a unique name.',
+}
+
+const customRecordGetters: RestrictedTypeGetters = {
+  getChangeRestrictedField: change => getChangeNestedField(change, SCRIPT_ID),
+  getSourceRestrictedFields: getCustomRecordRestrictedData,
+  getMessage: () => 'A Custom Record Type Field with that ID already exists',
+  getDetailedMessage: field => `Can't deploy the Custom Record Type Field: ${field} because there is already a Custom Record Type Field with that ID in the target environment.`
+    + ' Please select a Custom Record Type Field with a unique ID.',
+}
+
+const restrictedTypesGettersMap: Record<RestrictedTypes, RestrictedTypeGetters> = {
+  savedSearch: savedSearchGetters,
+  financialLayout: financialLayoutGetters,
+  customRecordField: customRecordGetters,
+}
+
+const getRestrictedType = (elemID: ElemID, isChange = true): RestrictedTypes | undefined => {
+  if (elemID.idType === INSTANCE_TYPE) {
+    if (elemID.typeName === SAVED_SEARCH) {
+      return SAVED_SEARCH_TYPE
```

Comment:
example of [this](https://github.com/salto-io/salto/pull/4189/files#r1165054318) comment
```suggestion
      return 'savedSearch'
```